discovery_engine_core/tooling/bin/news_client.rs

```diff
@@ -23,11 +35,63 @@ async fn main() -> Result<()> {
                   The token can be found in 1Password",
     )?;
 
+    let args: Vec<String> = env::args().collect();
+    let mode = args
+        .get(1)
+        .map(|s| s.to_owned())
+        .unwrap_or_else(|| "".to_string());
+
+    if mode == "newscatcher" {
```

Comment:
:art: limited number of if-else branches is usually ok, though this might look a bit cleaner with a `match mode {}`